cmd/finch/version.go

```diff
@@ -5,26 +5,53 @@ package main
 
 import (
 	"fmt"
+	"os/exec"
 
 	"github.com/spf13/cobra"
 
+	"github.com/runfinch/finch/pkg/command"
+	"github.com/runfinch/finch/pkg/flog"
+	"github.com/runfinch/finch/pkg/lima"
 	"github.com/runfinch/finch/pkg/version"
 )
 
-var finchVersion = version.Version
-
-func newVersionCommand() *cobra.Command {
+func newVersionCommand(limaCmdCreator command.LimaCmdCreator, logger flog.Logger) *cobra.Command {
 	versionCommand := &cobra.Command{
 		Use:   "version",
 		Args:  cobra.NoArgs,
 		Short: "Shows Finch version information",
-		RunE:  versionAction,
+		RunE:  newVersionAction(limaCmdCreator, logger).runAdapter,
 	}
-
+	
 	return versionCommand
 }
 
-func versionAction(cmd *cobra.Command, args []string) error {
-	fmt.Println("Finch version:", finchVersion)
+type versionAction struct {
+	creator command.LimaCmdCreator
+	logger  flog.Logger
+}
+
+func newVersionAction(creator command.LimaCmdCreator, logger flog.Logger) *versionAction {
+	return &versionAction{creator: creator, logger: logger}
+}
+
+func (va *versionAction) runAdapter(cmd *cobra.Command, args []string) error {
+	return va.printVersion()
+}
+
+func (va *versionAction) printVersion() error {
+	_, err := lima.GetVMStatus(va.creator, va.logger, limaInstanceName)
+	if err != nil {
+		fmt.Println("Finch version:", version.Version)
+	}
+
+	v, _ := exec.Command("lima", "nerdctl", "version", "--format", "json").Output()
```

Comment:
I think you can just use the `va.creator` again, instead of shelling out with exec:

```suggestion
        limaArgs := []string{"shell", limaInstanceName, "nerdctl", "version", "--format", "json"}
	out, err := va.creator.CreateWithoutStdio(limaArgs...).Output()
```